test/opt/instruction_test.cpp

```diff
@@ -478,6 +482,97 @@ TEST_F(DescriptorTypeTest, StorageBuffer) {
   EXPECT_FALSE(object_copy->IsReadOnlyPointer());
 }
 
+TEST_F(DescriptorTypeTest, ReadOnlyStorageBuffer) {
+  const std::string text = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %2 "main"
+               OpExecutionMode %2 OriginUpperLeft
+               OpSource GLSL 430
+               OpDecorate %3 DescriptorSet 0
+               OpDecorate %3 Binding 0
+               OpDecorate %4 BufferBlock
+               OpDecorate %4 NonWritable
+               OpDecorate %5 DescriptorSet 0
+               OpDecorate %5 Binding 1
+               OpDecorate %6 BufferBlock
+          %7 = OpTypeVoid
+          %8 = OpTypeFunction %7
+          %9 = OpTypeFloat 32
+         %10 = OpTypeVector %9 4
+         %11 = OpTypeRuntimeArray %10
+          %4 = OpTypeStruct %11
+          %6 = OpTypeStruct %11
+         %12 = OpTypePointer Uniform %4
+         %13 = OpTypePointer Uniform %6
+          %3 = OpVariable %12 Uniform
+          %5 = OpVariable %13 Uniform
+          %2 = OpFunction %7 None %8
+         %14 = OpLabel
+         %15 = OpCopyObject %11 %3
+               OpReturn
+               OpFunctionEnd
+)";
+
+  std::unique_ptr<IRContext> context =
+      BuildModule(SPV_ENV_UNIVERSAL_1_2, nullptr, text);
+  Instruction* type = context->get_def_use_mgr()->GetDef(12);
+  EXPECT_FALSE(type->IsVulkanStorageImage());
+  EXPECT_FALSE(type->IsVulkanSampledImage());
+  EXPECT_FALSE(type->IsVulkanStorageTexelBuffer());
+  EXPECT_TRUE(type->IsVulkanStorageBuffer());
+  EXPECT_FALSE(type->IsVulkanUniformBuffer());
+  EXPECT_TRUE(type->IsVulkanStorageBufferNonWritable());
+
+  Instruction* variable = context->get_def_use_mgr()->GetDef(3);
+  EXPECT_TRUE(variable->IsReadOnlyPointer());
+
+  Instruction* object_copy = context->get_def_use_mgr()->GetDef(12);
+  EXPECT_FALSE(object_copy->IsReadOnlyPointer());
+}
+
+TEST_F(DescriptorTypeTest, AliasedNotReadOnlyStorageBuffer) {
+  const std::string text = R"(
+               OpCapability Shader
+          %1 = OpExtInstImport "GLSL.std.450"
+               OpMemoryModel Logical GLSL450
+               OpEntryPoint Fragment %2 "main"
+               OpExecutionMode %2 OriginUpperLeft
+               OpSource GLSL 430
+               OpDecorate %3 DescriptorSet 0
+               OpDecorate %3 Binding 0
+               OpDecorate %4 BufferBlock
+               OpDecorate %4 NonWritable
```

Comment:
NonWritable applies to memory object declarations: in this case that means the OpVariable, not the struct type.